app/validators/presence.js

```diff
@@ -21,14 +23,27 @@ export default Base.extend({
   },
 
   validate(value, options) {
-    if (options.presence === true && isEmpty(value)) {
+    if (options.presence === true && !this._isPresent(value)) {
       return this.createErrorMessage('blank', value, options);
     }
 
-    if(options.presence === false && !isEmpty(value)) {
+    if(options.presence === false && this._isPresent(value)) {
       return this.createErrorMessage('present', value, options);
     }
 
     return true;
+  },
+
+  /**
+   * Handle presence of ember-data promise proxy based instances
+   */
+  _isPresent(value) {
+    if(typeOf(value) === 'instance' && value.constructor) {
+      const type = value.constructor.toString();
+      if(['DS.PromiseObject', 'DS.PromiseArray'].indexOf(type) !== -1) {
```

Comment:
ArrayProxy and ObjectProxy would handle both cases since DS.PromiseObject and DS.PromiseArray extend from them.  This would cover the case of comparing an object which is of those 4 types.

As for comparing, I'd lean towards using `detect` which will walk the inheritance graph.

`Ember.ArrayProxy.detect(foo); => true/false`